### ISSUE TITLE
Drop compatibility for Node versions < 4.0.0

### ISSUE THREAD

**Issue by meew0** [hydrabolt/discord.js#149]
With the recent influx of issues that are due to certain ES6 features not being available on node 0.12 (#148, #139) I think that instead of making polyfills for all of these, we should just drop node 0.12 compatibility altogether. Are there any valid reasons against this?

**Comment by meew0** [comment]
As of #150 and b3d5197, these issues should be fixed in indev. However, other issues like this will probably pop up, and I still think this is necessary.

**Comment by nicholastay** [comment]
Well, for me and a friend we've both been having issues playing audio on node v4+. We've been having to stay on node v0.12.x which is unfortunate but was the only solution for it to work.

I am on Windows but on my Mac I've briefly tested the same code and it works on node v4.2.0 so worst case I could do that. Friend however is on Linux and it doesn't work properly on node v4+, so it is some weird behavior. 

Basically nothing plays and nothing crashes, just nothing happens on playing audio.

**Comment by abalabahaha** [comment]
@nicholastay mind [hopping on the API server](https://discord.gg/0SBTUU1wZTYcFtmP) so it's easier to figure things out? As people have gotten everything working with Node 4, it's probably something your friend is doing wrong

**Comment by nicholastay** [comment]
Ok so I've resolved the issue by fixing problems with `node-opus` (reinstall/rebuild), so yeah no problem for me anymore. I think it is safe to ditch `v0.12.x` support.

**Comment by abalabahaha** [comment]
oh and also, the webdist does not appear to work, so either fix that or just drop webdist/node<4

**Comment by hydrabolt** [comment]
I agree. I think ditching `v0.12.x` would just reduce the amount of bugs and would also remove the need for Babel and Grunt.

**Comment by abalabahaha** [comment]
On second thought, Node 4 does not appear to have default parameter values by default (harmony flag needed)
